packages/core/elasticsearch/core-elasticsearch-server-internal/src/elasticsearch_service.ts

```diff
@@ -107,6 +106,13 @@ export class ElasticsearchService
       internalClient: this.client.asInternalUser,
     }).pipe(takeUntil(this.stop$));
 
+    // Log every error we may encounter in the connection to Elasticsearch
+    esNodesCompatibility$.subscribe(({ isCompatible, message }) => {
+      if (!isCompatible && message) {
+        this.log.error(message);
+      }
+    });
```

Comment:
IMO, these changes have a side effect: it will start polling the ES nodes earlier.

Observables don't run until they have a subscriber. And we are setting the subscriber sooner now.

I don't think it's necessarily a bad thing. Just highlighting the implications.

Reply:
Yeah that's true. I figured it's OK since we know we have the `this.client` already available.

Do you think we may be increasing error log volume too much? I was considering a specialised logger for these errors. Maybe `elasticsearch-service,nodes-compatibility-status`. Could help filter out these messages when debugging... WDYT?

Reply:
FWIW, functionally, I don't think we're changing anything.
The status service subscribes to the status emitter of this service since the setup lifecycle, so the client starts polling during the setup step. With these changes, it's no longer a side effect, it's intentional.

On top of that, AFAIK, we have a `distinct` filter buried in the emitter. So, if the error message doesn't change, we won't log again on every poll.